api/tests/integration/scripts/prod/delete-and-anonymise-organization-learners_test.js

```diff
@@ -0,0 +1,237 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import { DeleteAndAnonymiseOrgnizationLearnerScript } from '../../../../scripts/prod/delete-and-anonymise-organization-learners.js';
+import { databaseBuilder, knex } from '../../../test-helper.js';
+
+describe('DeleteAndAnonymiseOrgnizationLearnerScript', function () {
+  describe('Options', function () {
+    it('has the correct options', function () {
+      const script = new DeleteAndAnonymiseOrgnizationLearnerScript();
+      const { options } = script.metaInfo;
+
+      expect(options.organizationLearnerIds).to.deep.include({
+        type: '<array>number',
+        describe: 'a list of comma separated organization learner ids',
+        demandOption: true,
+      });
+      expect(options.organizationLearnerIds.coerce).to.be.a('function');
+    });
+
+    it('parses list of organizationLearnerIds', async function () {
+      const ids = '1,2,3';
+      const script = new DeleteAndAnonymiseOrgnizationLearnerScript();
+      const { options } = script.metaInfo;
+      const parsedData = await options.organizationLearnerIds.coerce(ids);
+
+      expect(parsedData).to.deep.equals([1, 2, 3]);
+    });
+  });
+
+  describe('Handle', function () {
+    let script;
+    let logger;
+    const ENGINEERING_USER_ID = 99999;
+
+    beforeEach(async function () {
+      script = new DeleteAndAnonymiseOrgnizationLearnerScript();
+      logger = { info: sinon.spy(), error: sinon.spy() };
+      sinon.stub(process, 'env').value({ ENGINEERING_USER_ID });
+    });
+
+    describe('anonymise organization learners', function () {
+      let learner, otherLearner, campaign, organization, clock, now;
+
+      beforeEach(async function () {
+        now = new Date('2024-01-17');
+        clock = sinon.useFakeTimers({ now, toFake: ['Date'] });
+
+        databaseBuilder.factory.buildUser({ id: ENGINEERING_USER_ID });
+        organization = databaseBuilder.factory.buildOrganization();
+        campaign = databaseBuilder.factory.buildCampaign({ organizationId: organization.id });
+        learner = databaseBuilder.factory.prescription.organizationLearners.buildOrganizationLearner({
+          id: 123,
+          organizationId: organization.id,
+        });
+        otherLearner = databaseBuilder.factory.prescription.organizationLearners.buildOrganizationLearner({
+          id: 456,
+          organizationId: organization.id,
+        });
+
+        await databaseBuilder.commit();
+      });
+
+      afterEach(function () {
+        clock.restore();
+      });
+
+      it('delete organization learners', async function () {
+        // given
+        databaseBuilder.factory.prescription.organizationLearners.buildOrganizationLearner({
+          id: 456,
+          organizationId: organization.id,
```

Comment:
**thought** : cette ligne ne sert pas. Elle est écrite plus haut dans le beforeEach, et n'est de toute façon pas commit dans ce it